packages/core-data/src/entities.js

```diff
@@ -133,6 +135,27 @@ export const kinds = [
 	{ name: 'taxonomy', loadEntities: loadTaxonomyEntities },
 ];
 
+/**
+ * Returns a function to be used to retrieve extra edits to apply before persisting a menu item.
+ *
+ * @param {Object} persistedRecord Already persisted Menu item
+ * @param {Object} edits           Edits.
+ * @return {Object} Updated edits.
+ */
+function prePersistMenuItem( persistedRecord, edits ) {
+	const newEdits = {
+		...persistedRecord,
+		...edits,
+	};
+	if ( Array.isArray( newEdits.menus ) ) {
+		newEdits.menus = newEdits.menus[ 0 ];
+	}
+	if ( newEdits.type === '' ) {
+		newEdits.type = 'custom';
+	}
+	return newEdits;
+}
```

Comment:
This ensures that all the menu item fields are always sent in the PUT request – ideally we wouldn't need it. I have a hunch that solving the `invalid_menu_order` situation could also help us here.

Reply:
It was agreed to just remove the validation. Which is why I worked on #34672. Once that PR is merged, rebase and you should be good to go.